hazelcast/src/main/java/com/hazelcast/client/impl/ClientHeartbeatMonitor.java

```diff
@@ -92,9 +92,9 @@ private void cleanupEndpointsWithDeadConnections() {
     }
 
     private void monitor(ClientEndpoint clientEndpoint) {
-        // C++ client sends heartbeat over its non-owner connections
-        // For other client types, disregard non-owner connections for heartbeat monitoring purposes
-        if (clientEndpoint.isOwnerConnection() == ClientType.CPP.equals(clientEndpoint.getClientType())) {
+        // C++ client does not send heartbeat over its owner connection
+        // We are skipping checking heartbeat for cpp owner connection
+        if (clientEndpoint.isOwnerConnection() && ClientType.CPP.equals(clientEndpoint.getClientType())) {
```

Comment:
The CPP client works with a single connection since CPP client 3.10, and hence can you add version check to this condition?

if (cppClientVersion < 3.10 or noClientVersionExist....